src/dom.js

```diff
@@ -87,14 +88,36 @@ export function waitForChildPromise(parent, checkFunc) {
 }
 
 /**
- * Waits for document's body to be available.
+ * Waits for document's head to be available.
+ * @param {!Document} doc
+ * @param {function()} callback
+ */
+export function waitForHead(doc, callback) {
+  waitForChild(doc.documentElement, () => !!doc.body, callback);
+}
+
+/**
+ * Waits for the document's head to be available.
+ * @param {!Document} doc
+ * @return {!Promise}
+ */
+export function waitForHeadPromise(doc) {
+  return new Promise(resolve => {
+    waitForHead(doc, resolve);
+  });
+}
+
+/**
+ * Waits for document's body to be available and ready.
  * Will be deprecated soon; use {@link AmpDoc#whenBodyAvailable} or
  * @{link DocumentState#onBodyAvailable} instead.
  * @param {!Document} doc
  * @param {function()} callback
  */
 export function waitForBody(doc, callback) {
-  waitForChild(doc.documentElement, () => !!doc.body, callback);
+  waitForChild(doc.documentElement, () => (
+    isDocumentReady(doc) && !!doc.body
```

Comment:
Ok, it appears this is failing tests for some obscure reasons (MutationObserver inside `waitForChild` doesn't fire during parsing). The fix is to just call `onDocumentReady` from `src/document-ready.js`. We can also change `waitForBodyPromise` to just call `whenDocumentReady`.

Reply:
oh okay, will use those functions 👍